modules/distributor/distributor.go

```diff
@@ -581,12 +578,21 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 	return keys, traces, nil
 }
 
-func countDiscaredSpans(numSuccessByTraceIndex []int, lastErrorReasonByTraceIndex []tempopb.PushErrorReason, traces []*rebatchedTrace, repFactor int) (maxLiveDiscardedCount, traceTooLargeDiscardedCount, unknownErrorCount int) {
+// discardedPredicate determines if a trace is discarded based on the number of successful replications.
+type discardedPredicate func(int) bool
+
+func newDiscardedPredicate(repFactor int) discardedPredicate {
```

Comment:
i'm curious why you chose this pattern. it's not a blocker, but i suppose i would have just written

```
func wasDiscarded(numSuccess int, repFactor int) bool {
  // the stuff
}
```

Reply:
It felt most natural to me not to pull the computation of `quorum` into the loop. Plus, having functions with the signature `(int, int)` is always a bit harder to read, as you need to inspect the parameter names more carefully. But it is certainly a matter of taste. Your solution is less code on the plus side. I can change it, if you prefer less code.

Reply:
i almost always prefer less code :)

but i also like different opinions, thoughts and patterns. leave it. it reminds me of function currying from my old js days.